client/app/states/dashboard/dashboard.state.js

```diff
@@ -129,24 +173,31 @@
 
       if (definedServiceIdsServices.subcount > 0) {
         vm.servicesCount.total = definedServiceIdsServices.subcount;
-        vm.servicesCount.current = definedServiceIdsServices.subcount === 0 ? nonRetiredServices.count :
-        retiredServices.subcount + nonRetiredServices.subcount;
 
-        vm.servicesCount.retired = vm.servicesCount.total - vm.servicesCount.current;
+        vm.servicesCount.retired = retiredServices.subcount;
 
         vm.servicesCount.soon = expiringServices.subcount;
+        
+        vm.servicesCount.current = vm.servicesCount.total - vm.servicesCount.retired - vm.servicesCount.soon;
       }
 
       vm.servicesFeature = true;
     }
 
     vm.requestsFeature = false;
-    if (angular.isDefined(pendingRequests)) {
+
+    if (angular.isDefined(pendingRequests) &&
+        angular.isDefined(approvedRequests) &&
+        angular.isDefined(deniedRequests)) {
       vm.requestsCount = {};
-      vm.requestsCount.total = pendingRequests.subcount + approvedRequests.subcount + deniedRequests.subcount;
-      vm.requestsCount.pending = pendingRequests.subcount;
-      vm.requestsCount.approved = approvedRequests.subcount;
-      vm.requestsCount.denied = deniedRequests.subcount;
+      vm.requestsCount.total = 0;
+
+      var allRequests = [pendingRequests, approvedRequests, deniedRequests];
+      var allRequestTypes = ['pending', 'approved', 'denied'];
+
+      lodash.times(3, function(n) {
```

Comment:
It isn't immediately clear the the 3 is the number of things in the array, maybe something like `allRequests.forEach(function(promises, idx) { ...`?

Or you could make it an object, like `{ pending: pendingRequests, ... }` and iterate over `_.pairs(obj)`?